### PR TITLE
Bump Jackson to 2.13.0

### PR DESCRIPTION
**Pull request by basil**
Amends #94.

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
I confirmed that earlier versions of the plugin were bundling jakarta.activation-api 1.2.1 as a transitive dependency.  The added dependency on  jakarta.activation-api 1.2.2 makes it clear that it is intentionally bundled.

**Review by timja — APPROVED** [review]

**Comment by jtnord** [comment]
> The added dependency on jakarta.activation-api 1.2.2 makes it clear that it is intentionally bundled.

there is no added `dependency` - it is a `dependencyManageent`  core (`2.303.1`ships `1.2.2` and as such if this is bundled without using mask-classes or plugin-first classloaded then it will not be used correctly and can even cause weird classloading issues if there are newly introduced classes. 
If the dependency is needed then the classloader needs to be tweaked, otherwise it should be excluded, unless I am missing something

File: pom.xml [hunk 1]
```diff
@@ -94,6 +94,11 @@
         <scope>import</scope>
         <type>pom</type>
       </dependency>
+      <dependency>
+        <groupId>jakarta.activation</groupId>
+        <artifactId>jakarta.activation-api</artifactId>
+        <version>1.2.2</version>
+      </dependency>
```
Comment (jtnord) [inline pom.xml:101]: jenkins ships with a lesser version.  if this is bundled in the plugin it is never used as the plugin is not using a plugin first classloader, so the fix should be to exclude the version instead.
if it is not bundled - then this means the test classpaths are more likely to more even invalid that normal.

**Review by jtnord — CHANGES_REQUESTED** [review]

**Comment by MarkEWaite** [comment]
> > The added dependency on jakarta.activation-api 1.2.2 makes it clear that it is intentionally bundled.
> 
> there is no added `dependency` - it is a `dependencyManageent` core (`2.303.1`ships `1.2.2` and as such if this is bundled without using mask-classes or plugin-first classloaded then it will not be used correctly and can even cause weird classloading issues if there are newly introduced classes. If the dependency is needed then the classloader needs to be tweaked, otherwise it should be excluded, unless I am missing something

I can't speak to the question whether the dependency is needed or not.  I've confirmed that `WEB-INF/lib/jakarta.activation-api-1.2.1.jar` is included in jackson2-api plugin 2.12.4.  The inclusion of `WEB-INF/lib/jakarta.activation-api-1.2.2.jar` as a transitive dependency in the jackson2-api plugin 2.13.0 release seems consistent with the previous release.

**Comment by jtnord** [comment]
https://projects.eclipse.org/projects/ee4j.jaf/releases/1.2.1

> First release from Eclipse. No API or implementation changes, just new licensing and Maven coordinates.

File: pom.xml [hunk 2]
```diff
@@ -68,7 +68,7 @@
     <changelist>-SNAPSHOT</changelist>
     <java.level>8</java.level>
     <jenkins.version>2.222.4</jenkins.version>
-    <jackson.version>2.12.4</jackson.version>
+    <jackson.version>2.13.0</jackson.version>
```
Comment (jtnord) [inline pom.xml:71]: should also change `revision` to keep in step  (not sure why we do not set that itself to `$jackon.version` but likely it does not work when it comes to release time.

**Comment by basil** [comment]
As @markewaite wrote, this is a preëxisting issue on the main branch that is unrelated to this PR. Get the main branch in order before you request changes of others.

**Comment by jtnord** [comment]
> Get the main branch in order before you request changes of others.

is offensive to me.

I was about to send a PR to your fork with the fixes, but how this is coming across is not helpful to a healthy community.

**Comment by timja** [comment]
What was the suggested fix? excluding the dependency, upgrading it in jenkins core? or something else?

**Comment by jtnord** [comment]
> What was the suggested fix? excluding the dependency, upgrading it in jenkins core? or something else?

it would depend on what changed API wise.  As per the eclipse release notes, what I am doing is excluding it (given if it currently works when bundled, it must work when not bundled as the plugin is not using plugin-first classloader or masking classes).
the version was bumped from the old activation-hudson-1.1.1 in https://github.com/jenkinsci/jenkins/pull/4660  so this is also bumping the baseline to we get the 1.2.1 version so there can not be conflicts between the 1.1 and 1.2 lines).

But anyway - the `version` of the plugin can not be `2.12.5-SNAPSHOT` when the api is at `2.13.0`  there is a PR incoming anyway to fix the bundling - but that is a 1:1 like for like and not bumping jackson in anyway shape or form

**Pull request closed without merging**